### PR TITLE
Add SAML 2.0 Metadata Refresh Sample

### PR DESCRIPTION
**Pull request by marcusdacoregio**
Closes gh-2

### CODE REVIEW

File: servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		return this.relyingPartyRegistrations.get(registrationId);
+	}
+
+	@Override
+	public Iterator<RelyingPartyRegistration> iterator() {
+		return this.relyingPartyRegistrations.values().iterator();
+	}
+
+	@Override
+	public void forEach(Consumer<? super RelyingPartyRegistration> action) {
```
Comment (jzheaux) [inline servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java:68]: What does implementing `forEach` and `spliterator` buy you? It seems like the default implementations do the same thing already.

File: servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	@Override
+	public Spliterator<RelyingPartyRegistration> spliterator() {
+		return this.relyingPartyRegistrations.values().spliterator();
+	}
+
+	@Scheduled(fixedDelay = 30, timeUnit = TimeUnit.MINUTES)
+	public void refreshMetadata() {
+		RelyingPartyRegistration metadata = RelyingPartyRegistrations
+				.fromMetadataLocation("https://simplesaml-for-spring-saml.apps.pcfone.io/saml2/idp/metadata.php")
```
Comment (jzheaux) [inline servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java:80]: It seems a little odd that you are taking some values from the properties file while other values are hard-coded into this method.

You might consider having this component depend on `Saml2RelyingPartyProperties` instead. Then you can get access to the relying party credentials but also the asserting party details.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @marcusdacoregio! I've left some inline feedback.

File: servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java (hunk 1 above)
Comment (marcusdacoregio) [reply servlet/spring-boot/java/saml2/refreshable-metadata/src/main/java/example/RefreshableRelyingPartyRegistrationRepository.java]: Probably an oversight. Didn't notice that there was default implementations.

**Comment by marcusdacoregio** [comment]
Thanks for the review @jzheaux. I've updated the PR.

**Review by jzheaux — APPROVED** [review]

**Pull request merged**